core/admin/EE_Admin_Page.core.php

```diff
@@ -632,13 +647,21 @@ private function _do_other_page_hooks()
                                );
                 throw new EE_Error(implode('||', $error_msg));
             }
-            $a = new ReflectionClass($classname);
-            // notice we are passing the instance of this class to the hook object.
-            $hookobj[] = $a->newInstance($this);
+            // // notice we are passing the instance of this class to the hook object.
+            $this->loader->getShared($classname, [$this]);
```

Comment:
using loader instead of Reflection so that dependencies can be auto loaded
also removed `$hookobj[]` because nothing was done with it